web/annotation_editor_layer_builder.css

```
@@ -188,6 +190,99 @@
       border: var(--focus-outline-around);
     }
   }
+
+  & .editToolbar {
+    --editor-toolbar-bin-image: url(images/editor-toolbar-bin.svg);
+    --editor-toolbar-bg-color: #f0f0f4;
+    --editor-toolbar-fg-color: #5b5b66;
+    --editor-toolbar-border-color: #8f8f9d;
+    --editor-toolbar-hover-bg-color: #e0e0e6;
+    --editor-toolbar-active-bg-color: #cfcfd8;
+    --editor-toolbar-focus-outline-color: #0060df;
+    --editor-toolbar-shadow: 0 2px 6px 0 rgb(58 57 68 / 0.2);
+
+    @media (prefers-color-scheme: dark) {
+      --editor-toolbar-bg-color: #2b2a33;
+      --editor-toolbar-fg-color: #fbfbfe;
+      --editor-toolbar-border-color: #2b2a33;
+      --editor-toolbar-hover-bg-color: #52525e;
+      --editor-toolbar-active-bg-color: #5b5b66;
+      --editor-toolbar-focus-outline-color: #0df;
+    }
+
+    @media screen and (forced-colors: active) {
+      --editor-toolbar-bg-color: ButtonFace;
+      --editor-toolbar-fg-color: ButtonText;
+      --editor-toolbar-border-color: ButtonText;
+      --editor-toolbar-hover-bg-color: AccentColor;
+      --editor-toolbar-active-bg-color: ButtonFace;
+      --editor-toolbar-focus-outline-color: ButtonBorder;
+      --editor-toolbar-shadow: none;
+    }
+
+    display: flex;
+    width: fit-content;
+    height: 28px;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    cursor: default;
+
+    position: absolute;
+    inset-inline-end: 0;
+    inset-block-start: calc(100% + 6px);
+
+    border-radius: 4px;
+    background-color: var(--editor-toolbar-bg-color);
+    border: 1px solid var(--editor-toolbar-border-color);
+    box-shadow: var(--editor-toolbar-shadow);
+
+    &.hidden {
+      display: none;
+    }
+
+    & .icons {
```

In cases such as these, when there's a space between the ampersand and the rule, is the `&` actually necessary?

Looking at https://developer.mozilla.org/en-US/docs/Web/CSS/Nesting_selector#_nesting_selector_and_whitespace it seems that these cases could be written slightly shorter as follows instead, and still achieve the same result?
```suggestion
    .icons {
```

---

Cool, thank you for the tip.
Maybe the css linter could remove these useless `&` for us.

---

> Maybe the css linter could remove these useless `&` for us.

Unfortunately there doesn't appear to be any special rules in Stylelint for the CSS nesting feature, I'm guessing it's still too new.